src/main/java/com/poliscrypts/api/service/ContactService.java

```diff
@@ -0,0 +1,119 @@
+     */
+    public ExtendedGenericPojoResponse addNewContact(Contact contact) {
+        if (contactRepository.findContactByFirstNameAndLastName(contact.getFirstName(), contact.getLastName()).isPresent()) {
+            throw new ContactException(messageSource.getMessage("contact.alreadyExist", null, new Locale("en")), 400);
+        }
+        //CHECK THE TVA NUMBER CONSTRAINT FOR EMPLOYEE & FREELANCER CONTACT
+        ContactHelper.verifiedTvaConstraint(contactTypeRepository, contact, messageSource);
+        contact.setContactType(contactTypeRepository.findByType(contact.getContactType().getType()));
+        contactRepository.saveAndFlush(contact);
+        return new ExtendedGenericPojoResponse(contact);
+    }
+
+    /**
+     * Update contact
+     * @param contact payload body of the contact
+     * @return code, message, contact object
+     */
+    public ExtendedGenericPojoResponse updateContact(Contact contact) {
+        if (!contactRepository.findById(contact.getId()).isPresent()) {
+            throw new ContactException(messageSource.getMessage("contact.notExist", null, new Locale("en")), 400);
+        }
+        //CHECK THE TVA NUMBER CONSTRAINT FOR EMPLOYEE & FREELANCER CONTACT
+        ContactHelper.verifiedTvaConstraint(contactTypeRepository, contact, messageSource);
+        contact.setContactType(contactTypeRepository.findByType(contact.getContactType().getType()));
+        contactRepository.saveAndFlush(contact);
+        return new ExtendedGenericPojoResponse(contact);
+    }
+
+    /**
+     * Delete contact
+     * @param id contact id should be deleted
+     * @return code, message, contact object
+     */
+    public GenericPojoResponse deleteContact(Integer id) {
+        Optional<Contact> contact = contactRepository.findById(id);
+        if (!contact.isPresent()) {
+            throw new ContactException(messageSource.getMessage("contact.notExist", null, new Locale("en")), 400);
+        }
+        contactRepository.delete(contact.get());
+        return new GenericPojoResponse(0, messageSource.getMessage("SUCCESS", null, new Locale("en")));
+    }
+
+    /**
+     * Assign a contact into specific company
+     * @param contactId contact id to be assigned
+     * @param companyId company id
+     * @return code, message
+     */
+    public GenericPojoResponse assignContact2Company(Integer contactId, Integer companyId) {
+        Optional<Contact> contact = contactRepository.findById(contactId);
+        Optional<Company> company = companyRepository.findById(companyId);
+        if(!contact.isPresent()) {
+            throw new ContactException(messageSource.getMessage("contact.notExist", null, new Locale("en")), 400);
+        }
+        if(!company.isPresent()) {
+            throw new CompanyException(messageSource.getMessage("company.notExist", null, new Locale("en")), 400);
+        }
+        Contact pojoContact = contact.get();
+        List<Company> companies = pojoContact.getCompanies();
```

Comment:
check if contact have already the same company, if you already used a Set instead of List of companies your controller advice will have work to do :)